src/components/TransactionalInfo.tsx

```diff
@@ -60,14 +61,19 @@ const TransactionalInfo = ({
   const address = useAppSelector((state) => selectAddressByHash(state, addressHash))
   const { amount, direction, outputs, lockTime, infoType } = useTransactionInfo(tx, addressHash, showInternalInflows)
   const { label, amountTextColor, amountSign: sign, Icon, iconColor, iconBgColor } = useTransactionUI(infoType)
+  const isPending = isPendingTx(tx)
 
   if (!address) return null
 
-  const pendingToAddressComponent = isPendingTx(tx) && (
-    <AddressBadge truncate addressHash={tx.toAddress} showHashWhenNoLabel withBorders />
-  )
-  const amountSign =
-    showInternalInflows && infoType === 'move' && !isPendingTx(tx) && !isConsolidationTx(tx) ? '- ' : sign
+  const pendingToAddressComponent = isPending ? (
+    tx.type === 'contract' ? (
+      <Badge>{t('Smart contract')}</Badge>
+    ) : (
```

Comment:
I decided to go with a badge. I need to do more testing using a dapp to see if I can display something better.